src/main/java/ua/lviv/iot/searcher/Searcher.java

```diff
@@ -0,0 +1,35 @@
+package ua.lviv.iot.searcher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ua.lviv.iot.graph.Graph;
+
+public class Searcher {
+
+	public List<String> findOptimalSequenceFor(Graph<String> graph) {
+		int numberOfVertices = graph.numberOfVertices();
+		List<String> optimalSequence = new ArrayList<String>(numberOfVertices);
+		boolean[] isVisited = new boolean[numberOfVertices];
```

Comment:
By the way, just for the fun of it – do you know how much memory a boolean in Java takes?

The answer is here: https://passtor.blogspot.com/2009/11/boolean-java.html